### ISSUE TITLE
KeyError: 'LD_LIBRARY_PATH'

### ISSUE THREAD

**Issue by shivanandmn** [TimDettmers/bitsandbytes#560]
```
(llm) sigmoid@Sigmoid:/mnt/c/Users/shiva/Developer/ml-projects/llms/red_pajama$ python -m bitsandbytes                                                                                                                    
                                                                                                                                                                                                                          
===================================BUG REPORT===================================                                                                                                                                          
Welcome to bitsandbytes. For bug reports, please run                                                                                                                                                                      
                                                                                                                                                                                                                          
python -m bitsandbytes                                                                                                                                                                                                    
                                                                                                                                                                                                                          
 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues                                                                                                        
================================================================================                                                                                                                                          
bin /home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so                                                                                                                    
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication,
 and GPU quantization are unavailable.                                                                                                                                                                                    
  warn("The installed version of bitsandbytes was compiled without GPU support. "                                                                                                                                         
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so: undefined symbol: cadam32bit_grad_fp32                                                                                
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: /home/sigmoid/miniconda3/envs/llm did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.
0'] as expected! Searching further paths...                                                                                                                                                                               
  warn(msg)

/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('unix')}
  warn(msg)
CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. Searching in backup paths...
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('/usr/lo
cal/cuda/lib64')}
  warn(msg)
CUDA SETUP: WARNING! libcuda.so not found! Do you have a CUDA driver installed? If you are on a cluster, make sure you are on a CUDA machine!
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: No libcudart.so found! Install CUDA or the cudatoolkit package (anaconda)!
  warn(msg)
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: No GPU detected! Check your CUDA paths. Proceeding to load CPU-only library...
  warn(msg)
CUDA SETUP: Loading binary /home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so...
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
++++++++++++++++++ BUG REPORT INFORMATION ++++++++++++++++++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

+++++++++++++++++++ ANACONDA CUDA PATHS ++++++++++++++++++++
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda121_nocublaslt.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda120_nocublaslt.so




/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda114_nocublaslt.so                                                                                                  [0/574]
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda112.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda121.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda120.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda117.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/torch/lib/libtorch_cuda_linalg.so
/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/torch/lib/libc10_cuda.so

++++++++++++++++++ /usr/local CUDA PATHS +++++++++++++++++++


+++++++++++++++ WORKING DIRECTORY CUDA PATHS +++++++++++++++


++++++++++++++++++ LD_LIBRARY CUDA PATHS +++++++++++++++++++
Traceback (most recent call last):
  File "/home/sigmoid/miniconda3/envs/llm/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/sigmoid/miniconda3/envs/llm/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/__main__.py", line 95, in <module>
    generate_bug_report_information()
  File "/home/sigmoid/miniconda3/envs/llm/lib/python3.10/site-packages/bitsandbytes/__main__.py", line 66, in generate_bug_report_information
    lib_path = os.environ['LD_LIBRARY_PATH'].strip()
  File "/home/sigmoid/miniconda3/envs/llm/lib/python3.10/os.py", line 680, in __getitem__
    raise KeyError(key) from None
KeyError: 'LD_LIBRARY_PATH'




```

**Comment by papy-ganjha** [comment]
Hello!
You're cuda path seems not set. If you already have installed Cuda, check if you have something in the following folder: `/usr/local/cuda-XX.X` (don't forget to change the XX.X by the cuda version you installed).
If there is a folder you can simply add lines to set the path to you're `.bashrc` (or `.zshrc` if you use zsh) by doing so: `export LD_LIBRARY_PATH=/usr/local/cuda-XX.X/lib64`.
Restart you terminal and retry the command it should work.

**Comment by ht0rohit** [comment]
Hi, I'm also getting the same error. I've CUDA version 121. I'm doubtful that the package supports version 121 as the following is mentioned on the PyPy page -
`# CUDA_VERSIONS in {110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 120}`
Can someone please confirm that this may be the cause?

**Comment by TimDettmers** [comment]
It should support 121. What might have gone i your case @ht0rohit is that multiple CUDA versions are installed. It seems here no CUDA versions are installed and the LD_LIBRARY_PATH is set. As such, bitsandbytes cannot find CUDA and fails. 

Can you please try run the command `which nvcc`  and `nvidia-smi` to see if CUDA is installed and report the outputs back here.

**Comment by TimDettmers** [comment]
This has been fixed in f4996978db44102bc30b2e97820ab63d108c234c. Version 122 has been added and added to the documentation. Thank you for reporting this issue.

**Comment by swumagic** [comment]
Bitsandbytes was not supported windows before, but my method can support windows.（yuhuang）
1  open folder J:\StableDiffusion\sdwebui，Click the address bar of the folder and enter CMD
or WIN+R, CMD  。enter，cd /d J:\StableDiffusion\sdwebui
2  J:\StableDiffusion\sdwebui\py310\python.exe -m pip uninstall bitsandbytes

3 J:\StableDiffusion\sdwebui\py310\python.exe -m pip uninstall bitsandbytes-windows

4 J:\StableDiffusion\sdwebui\py310\python.exe -m pip install https://github.com/jllllll/bitsandbytes-windows-webui/releases/download/wheels/bitsandbytes-0.41.1-py3-none-win_amd64.whl

Replace your SD venv directory file（python.exe Folder） here（J:\StableDiffusion\sdwebui\py310）
